uqcsbot/starboard.py

```diff
@@ -1,38 +1,71 @@
 import os, time
 from threading import Timer
-from typing import Tuple, List
-from zoneinfo import ZoneInfo
+from typing import Tuple, List, Union
 
 import discord
 from discord import app_commands
 from discord.ext import commands
 from sqlalchemy.sql.expression import and_
 
 from uqcsbot import models
+from uqcsbot.bot import UQCSBot
 from uqcsbot.utils.err_log_utils import FatalErrorWithLog
 
 
-class BlacklistedMessageError(Exception):
-    # always caught. used to differentiate "starboard message doesn't exist" and "starboard message is blacklisted"
-    pass
+class StarboardMsgPair(object):
+    def __init__(
+        self,
+        recv: Union[discord.Message, None],
+        sent: Union[discord.Message, None],
+        recv_channel: Union[
+            discord.abc.GuildChannel, discord.Thread, discord.abc.PrivateChannel, None
+        ],
+        blacklist: bool = False,
+    ):
+        self.recv = recv
+        self.sent = sent
+        self.recv_channel: Union[discord.TextChannel, None] = (
+            recv_channel
+            if recv_channel is None or isinstance(recv_channel, discord.TextChannel)
+            else None
+        )
+        self.blacklist = blacklist
 
+    def dangerous_recv(self) -> discord.Message:
```

Comment:
I didn't understand the `dangerous` commands on first run through; a doc string would be nice but not necessary.